squidpy/pl/_spatial.py

```diff
@@ -0,0 +1,415 @@
+    scalebar_dx: _SeqFloat = None,
+    scalebar_units: _SeqStr = None,
+    # title and axis
+    title: _SeqStr = None,
+    axis_label: _SeqStr = None,
+    fig: Figure | None = None,
+    ax: Axes | Sequence[Axes] | None = None,
+    figsize: tuple[float, float] | None = None,
+    dpi: int | None = None,
+    save: str | Path | None = None,
+    # kwargs
+    scalebar_kwargs: Mapping[str, Any] = MappingProxyType({}),
+    edges_kwargs: Mapping[str, Any] = MappingProxyType({}),
+    **kwargs: Any,
+) -> None:
+    """
+    Plot spatial omics data saved in AnnData.
+
+    Parameters
+    ----------
+    %(adata)s
+    %(spatial_key)s
+    %(shape)s
+    %(color)s
+    %(groups)s
+    %(library_id)s
+    %(library_key)s
+    %(plotting_image)s
+    %(plotting_segment)s
+    %(plotting_features)s
+    scalebar_kwargs
+        Keyword arguments for :func:`matplotlib_scalebar.ScaleBar`.
+    edges_kwargs
+        Keyword arguments for :func:`networkx.draw_networkx_edges`.
+    kwargs
+        Keyword arguments for :func:`matplotlib.pyplot.scatter` or :func:`matplotlib.pyplot.imshow`.
+
+    Returns
+    -------
+    %(plotting_returns)s
+    """
+    _sanitize_anndata(adata)
+    _assert_spatial_basis(adata, spatial_key)
+
+    scalebar_kwargs = dict(scalebar_kwargs)
+    edges_kwargs = dict(edges_kwargs)
+
+    color_params = _prepare_args_plot(
+        adata=adata,
+        shape=shape,
+        color=color,
+        groups=groups,
+        alpha=alpha,
+        img_alpha=img_alpha,
+        use_raw=use_raw,
+        layer=layer,
+        palette=palette,
+    )
+
+    spatial_params = _image_spatial_attrs(
+        adata=adata,
+        shape=shape,
+        spatial_key=spatial_key,
+        library_id=library_id,
+        library_key=library_key,
+        img=img,
+        img_res_key=img_res_key,
+        img_channel=img_channel,
+        seg=seg,
+        seg_key=seg_key,
+        cell_id_key=cell_id_key,
+        scale_factor=scale_factor,
+        size=size,
+        size_key=size_key,
+        img_cmap=img_cmap,
+    )
+
+    coords, crops = _set_coords_crops(
+        adata=adata,
+        spatial_params=spatial_params,
+        spatial_key=spatial_key,
+        library_key=library_key,
+        crop_coord=crop_coord,
+    )
+
+    fig_params, cmap_params, scalebar_params, kwargs = _prepare_params_plot(
+        color_params=color_params,
+        spatial_params=spatial_params,
+        spatial_key=spatial_key,
+        wspace=wspace,
+        hspace=hspace,
+        ncols=ncols,
+        cmap=cmap,
+        norm=norm,
+        library_first=library_first,
+        img_cmap=img_cmap,
+        frameon=frameon,
+        na_color=na_color,
+        title=title,
+        axis_label=axis_label,
+        scalebar_dx=scalebar_dx,
+        scalebar_units=scalebar_units,
+        dpi=dpi,
+        figsize=figsize,
+        fig=fig,
+        ax=ax,
+        **kwargs,
+    )
+
+    _subset: Callable[[AnnData, str | None, str | None], AnnData] = (
```

Comment:
Tbh don't like this that much, saves you almost nothing (+ have to ignore types).
Would just later use `adata_sub = adata if library_key is None else _subs(adata, library_key=library_key, library_id=_id)` (low prio)

Reply:
I use it twice though thought it's nicer